app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/NavigationMenuEntriesGenerator.kt

```diff
@@ -1,45 +1,67 @@
- *
- * @param numDays Number of days. Must be 1 or more.
- * @param dateInfos A [list of DateInfo objects][DateInfos].
- *                  The [dayIdx] of the first object must be 1.
- *                  The list cannot be null nor empty.
- * @param currentDate A moment instance representing the day of interest.
- * @param todayString The word "Day" in the language of choice.
- * @param todayString The word "Today" in the language of choice.
- */
-internal fun getDayMenuEntries(numDays: Int, dateInfos: DateInfos?, currentDate: Moment, dayString: String, todayString: String): Array<String?> {
-    if (numDays < 1) {
-        throw IllegalArgumentException("Invalid number of days: $numDays")
-    }
-    if (dateInfos == null || dateInfos.isEmpty()) {
-        throw IllegalArgumentException("Invalid date info list: $dateInfos")
+internal class NavigationMenuEntriesGenerator @JvmOverloads constructor(
+
+    /**
+     * The word "Day" in the language of choice.
+     */
+    private val dayString: String,
+    /**
+     * The word "Today" in the language of choice.
+     */
+    private val todayString: String,
+
+    private val logging: Logging = Logging.get()
+
+) {
+
+    private companion object {
+        const val LOG_TAG = "NavigationMenuEntriesGenerator"
     }
-    val entries = arrayOfNulls<String>(numDays)
-    for (dayIndex in 0 until numDays) {
-        var entry = "$dayString ${dayIndex + 1}"
-        for (dateInfo in dateInfos) {
-            if (dateInfo.dayIdx == dayIndex + 1) {
-                if (currentDate == dateInfo.date) {
-                    entry += " - $todayString"
+
+    /**
+     * Returns a [String] array of day menu entries.
+     *
+     * If the [currentDate] matches a date in [dateInfos] then the corresponding day will be suffixed
+     * with the given [todayString]. Example: ["Day 1", "Day 2 - Today", "Day 3"]
+     * An [IllegalArgumentException] is thrown the parameter restrictions are not met.
+     *
+     * @param numDays Number of days. Must be 1 or more.
+     * @param dateInfos A [list of DateInfo objects][DateInfos]. The [dayIdx] of the first object
+     * must be 1. The list cannot be null nor empty.
+     * @param currentDate A moment instance representing the day of interest.
+     */
+    @JvmOverloads
+    fun getDayMenuEntries(
+        numDays: Int,
+        dateInfos: DateInfos?,
+        currentDate: Moment = Moment.now().startOfDay()
+    ): List<String> {
+        if (numDays < 1) {
+            throw IllegalArgumentException("Invalid number of days: $numDays")
+        }
+        if (dateInfos == null || dateInfos.isEmpty()) {
+            throw IllegalArgumentException("Invalid date info list: $dateInfos")
+        }
+        logging.d(LOG_TAG, "Today is " + currentDate.toUtcDateTime().toLocalDate())
+        val entries = mutableListOf<String>()
+        for (dayIndex in 0 until numDays) {
+            var entry = "$dayString ${dayIndex + 1}"
+            for (dateInfo in dateInfos) {
```

Comment:
I am think it might be good to have an additional method which simply converts DateInfos into an Array of DayOfTheWeek (aka an Integer). Then a simple check if this Array of DayOfTheWeek contains `dayIndex` would replace the entire inner loop.

Reply:
:+1: Sounds good to me. This should be something for a separate housekeeping pull request.

Reply:
Sure